astropy/modeling/core.py

```diff
@@ -806,38 +801,14 @@ def outputs(self, val):
         self._outputs = val
 
     @property
-    def n_inputs(self):
-        # TODO: remove the code in the ``if`` block when support
-        # for models with ``inputs`` as class variables is removed.
-        if hasattr(self.__class__, "n_inputs") and isinstance(
-            self.__class__.n_inputs, property
-        ):
-            try:
-                return len(self.__class__.inputs)
-            except TypeError:
-                try:
-                    return len(self.inputs)
-                except AttributeError:
-                    return 0
-
-        return self.__class__.n_inputs
+    def n_inputs(self) -> int:
+        """The number of inputs."""
+        return len(getattr(self, "inputs", ()))
```

Comment:
Should we not keep `return self.__class__.n_inputs`?

Reply:
I know that's what the comment said but empirically: no. That branch was actually never reached in any test.